docs/gallery/plot_dynamictable_howto.py

```diff
@@ -318,6 +318,34 @@
     columns=[dtr_idx, indexed_dtr_col],
 )
 
+###############################################################################
+# Setting the target table of a DynamicTableRegion column of a DynamicTable
+# -------------------------------------------------------------------------
+# A subclass of DynamicTable might have a pre-defined DynamicTableRegion column
+# that references rows of another table. The target table of this column might
+# not be initialized. In this case, the target table can be set using the
+# ``target_tables`` argument of the DynamicTable constructor. This argument
+# should be a dictionary mapping the name of the DynamicTableRegion column to
+# the target table. The target table can also be set after the DynamicTable
+# has been initialized using ``my_table.my_column.table = other_table``.
+
+class SubTable(DynamicTable):
+    __columns__ = (
+        {'name': 'dtr', 'description': 'required region', 'required': True, 'table': True},
+    )
+
+referenced_table = DynamicTable(
+    name='referenced_table',
+    description='an example table',
+)
+
+sub_table = SubTable(
+    name='sub_table',
+    description='an example table',
+    target_tables={'dtr': referenced_table},
+)
+# now the target table of the DynamicTableRegion column 'dtr' is set to `referenced_table`
+
```

Comment:
It would be useful to contrast this approach with how this would look when using the `columns` argument of `DynamciTable.__init__`. E.g., we could add a note to explain why this is useful. Maybe something like:

```suggestion
# now the target table of the DynamicTableRegion column 'dtr' is set to `referenced_table`

.. note::

     Using `target_tables` to specify the target means that we do not need to create 
     `DynamicTableRegion` columns that are specified in the schema manually, but  we can have
     `DynamicTable.__init__` create the column on our behalf. The alternative would be to create
     the `DynamicTableRegion` column and provide it as part of the `columns` argument of 
      `DynamicTable.__init__`, however, this means it is up to the user to ensure that the column is
      created in accordance to the schema of the specific `DynamicTable` type. 
     column  

```

Reply:
I tried to address this in the revised text above. Please review.